patches/0060-Create-Logging-Middleware-for-Lambda-service-that-do.patch

```diff
@@ -0,0 +1,158 @@
++		ctx = logger.SetField(ctx, string(logging.SigningRegionKey), signingRegion)
++	}
++	if awsmiddleware.GetEndpointSource(ctx) == aws.EndpointSourceCustom {
++		ctx = logger.SetField(ctx, string(logging.CustomEndpointKey), true)
++	}
++
++	req, ok := in.Request.(*smithyhttp.Request)
++	if !ok {
++		return out, metadata, fmt.Errorf("unexpected request middleware type %T", in.Request)
++	}
++
++	rc := req.Build(ctx)
++
++	originalBody := rc.Body
++	// remove the body from the logging output
++	redactedBody := strings.NewReader("[Redacted]")
++	rc.Body = io.NopCloser(redactedBody)
++	rc.ContentLength = redactedBody.Size()
++
++	requestFields, err := logging.DecomposeHTTPRequest(ctx, rc)
++	if err != nil {
++		return out, metadata, fmt.Errorf("decomposing request: %w", err)
++	}
++	logger.Debug(ctx, "HTTP Request Sent", requestFields)
++
++	// reconstruct the original request
++	req, err = req.SetStream(originalBody)
++	if err != nil {
++		return out, metadata, err
++	}
++	in.Request = req
++
++	start := time.Now()
++	out, metadata, err = next.HandleDeserialize(ctx, in)
++	duration := time.Since(start)
++
++	if err != nil {
++		return out, metadata, err
++	}
++
++	if res, ok := out.RawResponse.(*smithyhttp.Response); !ok {
++		return out, metadata, fmt.Errorf("unknown response type: %T", out.RawResponse)
++	} else {
++		responseFields, err := decomposeHTTPResponse(ctx, res.Response, duration)
++		if err != nil {
++			return out, metadata, fmt.Errorf("decomposing response: %w", err)
++		}
++		logger.Debug(ctx, "HTTP Response Received", responseFields)
++	}
++
++	return out, metadata, err
++}
+diff --git a/internal/service/lambda/service_package_extra.go b/internal/service/lambda/service_package_extra.go
+index 54f6aac15a..1f2440d3e3 100644
+--- a/internal/service/lambda/service_package_extra.go
++++ b/internal/service/lambda/service_package_extra.go
+@@ -6,6 +6,7 @@ import (
+ 	aws_sdkv2 "github.com/aws/aws-sdk-go-v2/aws"
+ 	retry_sdkv2 "github.com/aws/aws-sdk-go-v2/aws/retry"
+ 	lambda_sdkv2 "github.com/aws/aws-sdk-go-v2/service/lambda"
++	"github.com/aws/smithy-go/middleware"
+ 	tfawserr_sdkv2 "github.com/hashicorp/aws-sdk-go-base/v2/tfawserr"
+ 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+ 	"github.com/hashicorp/terraform-provider-aws/names"
+@@ -34,6 +35,19 @@ func (p *servicePackage) NewClient(ctx context.Context, config map[string]any) (
+ 		if endpoint := config[names.AttrEndpoint].(string); endpoint != "" {
+ 			o.BaseEndpoint = aws_sdkv2.String(endpoint)
+ 		}
++
++		// Switch out the terraform http logging middleware with a custom logging middleware that does not log the
++		// lambda code. Logging the lambda code leads to memory bloating because it allocates a lot of copies of the
++		// body
++		o.APIOptions = append(o.APIOptions, func(stack *middleware.Stack) error {
++			loggingMiddleware, err := stack.Deserialize.Remove("TF_AWS_RequestResponseLogger")
```

Comment:
If the name of the logging middleware ever changes it'll fail during the integration tests

Reply:
Nice !!